download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -177,37 +191,55 @@
     @Parameter(defaultValue = "${settings}", readonly = true, required = true)
     private Settings settings;
 
-    /**
-     * Method call whent he mojo is executed for the first time.
-     * @throws MojoExecutionException if an error is occuring in this mojo.
-     * @throws MojoFailureException if an error is occuring in this mojo.
-     */
-    public void execute() throws MojoExecutionException, MojoFailureException {
+	/**
+	 * Map of HTTP parameters to be used to download content. These properties are to be given as
+	 * &lt;httpParameters&gt;
+	 * 	&lt;aParameterName&gt;aParameterValue&lt;/aParameterName&gt;
+	 * &lt;httpParameters&gt;
+	 */
+    @Parameter(property = "parameters")
+	private final Map<String, String> parameters = new HashMap<>();
+
+	/**
+	 * Map of HTTP headers to be used to download content. These properties are to be given as
+	 * &lt;httpHeaders&gt;
+	 * 	&lt;aHeaderName&gt;aParameterValue&lt;/aHeaderName&gt;
+	 * &lt;/httpHeaders&gt;
+	 */
+    @Parameter(property = "headers")
+	private final Map<String, String> headers = new HashMap<>();
+
+	/**
+	  * Method call whent he mojo is executed for the first time.
+	  * @throws MojoExecutionException if an error is occuring in this mojo.
+	  * @throws MojoFailureException if an error is occuring in this mojo.
+	  */
+	public void execute() throws MojoExecutionException, MojoFailureException {
         if (this.skip) {
-            getLog().info("maven-download-plugin:wget skipped");
-            return;
-        }
-        
+			getLog().info("maven-download-plugin:wget skipped");
+			return;
+		}
+
         if (StringUtils.isNotBlank(serverId) && (StringUtils.isNotBlank(username) || StringUtils.isNotBlank(password))) {
             throw new MojoExecutionException("Specify either serverId or username/password, not both");
         }
-        
+
         if (settings == null) {
             getLog().warn("settings is null");
         }
         getLog().debug("Got settings");
-        if (retries < 1) {
-            throw new MojoFailureException("retries must be at least 1");
-        }
+		if (retries < 1) {
```

Comment:
Would be nice if you could avoid such whitespace changes as they don't bring anything and create noise (and even inconsistency with the current usage there). It would be entirely fine to replace spaces by tabs, but it would be better in another commit.
(note that this is not a blocker for this patch to be merged; as the feature is valuable we wouldn't make a whitespace change blocking it).